docs/architecture/devices-and-services.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Entities: integrating devices & services"
+sidebar_label: "Introduction"
+---
+
+Integrations can represent devices & services in Home Assistant. The data points are represented as entities. Entities are standardized by other integrations like `light`, `switch`, etc. Standardized entities come with services for control, but an integration can also provide their own services in case something is not standardized.
+
+An entity abstracts away the internal working of Home Assistant. As an integrator you don't have to worry about how services or the state machine work. Instead, you extend an entity class and implement the necessary properties and methods for the device type that you're integrating.
+
+<img className='invertDark'
+  src='/img/en/architecture/integrating-devices-services.svg'
+  alt='Integrating devices & services' />
+
+<!--
+  https://docs.google.com/drawings/d/1oysZ1VMcPPuyKhY4tequsBWcblDdLydbWxlu6bH6678/edit?usp=sharing
+-->
+
+Configuration is provided by the user via a [Config Entry](../config_entries_index.md) or in special/legacy cases via [configuration.yaml](../configuration_yaml_index.md).
+
+The device integration (i.e. `hue`) will use this configuration to set up a connection with the device/service. It will forward the config entry (legacy uses discovery helper) to set up its entities in their respective integrations (light, switch).
+
+The entity integration (i.e. `light`) is responsible for defining the abstract entity class and services to control the entities.
+
+The Entity Component helper is responsible for distributing the configuration to the platforms, forward discovery and collect entities for service calls.
+
+The Entity Platform helper manages all entities for the platform and polls them for updates if necessary. When adding entities, the Entity Platform is responsible for registering the entity with the device and entity registries.
+
+Integration Platform (i.e. `hue.light`) uses configuration to query the external device/service and create entities to be added. It is also possible for integration platforms to register entity services. These services will only target the platform entities (across all platforms that belong to the integration) and are hosted under the device integration domain (i.e.. hue.activate_scene).
```

Comment:
Maybe make it more clear that the entity service will target all platforms for the integration of the same domain, ie all hue.light but not eg hue.switch if registering a hue light service.